src/HypercertMinter.sol

```diff
@@ -140,6 +140,16 @@ contract HypercertMinter is IHypercertToken, SemiFungible1155, AllowlistMinter,
         _uri = super.uri(tokenID);
     }
 
+    /// TRANSFER RESTRICTIONS
+
+    function readTransferRestriction(uint256 tokenID) external view returns (string memory) {
+        TransferRestrictions temp = typeRestrictions[getBaseType(tokenID)];
+        if (temp == TransferRestrictions.AllowAll) return "AllowAll";
+        if (temp == TransferRestrictions.DisallowAll) return "DisallowAll";
+        if (temp == TransferRestrictions.FromCreatorOnly) return "FromCreatorOnly";
```

Comment:
This might be wishful thinking, but is there an "exhaustive checking" paradigm in Solidity? so that the compiler can tell you if you forgot to check for an enum (if we add one in the future)?

Rust does this automatically.
Typescript you can achieve this with the assertNever pattern.

Reply:
Unfortunately not, this is also why I think we should assist development by adding the mapping in the SDK